components/Snackbar/src/private/MDCSnackbarOverlayView.m

```diff
@@ -454,38 +464,55 @@ - (void)slideMessageView:(MDCSnackbarMessageView *)snackbarView
         toContentOpacity:(CGFloat)toContentOpacity
               completion:(void (^)(void))completion {
   // Prepare to move the snackbar.
-  _snackbarOnscreenConstraint.active = onscreen;
-  _snackbarOffscreenConstraint.active = !onscreen;
-  [_containingView setNeedsUpdateConstraints];
-
+  NSTimeInterval duration = MDCSnackbarLegacyTransitionDuration;
+  if (!snackbarView.message.usesLegacySnackbar) {
+    duration = onscreen ? MDCSnackbarEnterTransitionDuration : MDCSnackbarExitTransitionDuration;
+  }
   CAMediaTimingFunction *timingFunction =
-      [CAMediaTimingFunction mdc_functionWithType:MDCAnimationTimingFunctionEaseInOut];
+  [CAMediaTimingFunction mdc_functionWithType:MDCAnimationTimingFunctionEaseInOut];
   [CATransaction begin];
   [CATransaction setAnimationTimingFunction:timingFunction];
+  [CATransaction setCompletionBlock:completion];
+  [CATransaction setAnimationDuration:duration];
+  CAAnimationGroup *animationsGroup = [CAAnimationGroup animation];
+  animationsGroup.fillMode = kCAFillModeForwards;
+  animationsGroup.removedOnCompletion = NO;
+
+  if (snackbarView.message.usesLegacySnackbar) {
+    _snackbarOnscreenConstraint.active = onscreen;
+    _snackbarOffscreenConstraint.active = !onscreen;
+    [_containingView setNeedsUpdateConstraints];
+    // We use UIView animation inside a CATransaction in order to use the custom animation curve.
+    [UIView animateWithDuration:duration
+                          delay:0
+                        options:UIViewAnimationOptionCurveEaseInOut
```

Comment:
Perhaps pass 0 to options to indicate that we're not attempting to customize the timing function here.

Reply:
done